### DISCUSSION TITLE
Updating Docs

### DISCUSSION THREAD

**Discussion by ESRM-Support** [oqtane/oqtane.framework#2203] [Q&A] [answered]
I have been making quite a few changes to my local version of the Oqtane documentation as I try to understand how to use this awesome framework. Probably by next weekend I will have finished everything I wanted to document. I saw where it is best to post a PR to the oqtane.docs repo to see if it might be useful. But is there anyone else who is in the process of updating the docs to collaborate with right now, or should I just post what I have when I am finished?

**Comment by sbwalker** [comment, marked as answer]
@iJungleboy do you have any suggestions?

**Reply by iJungleboy** [reply]
@ESRM-Research thanks for helping!

I'm in charge of the docs, so I believe it would be best if you sync with me.

**Reply by ESRM-Support** [reply]
@iJungleboy Thanks for reaching out. I have forked the oqtane.docs and then updated them with the changes that I have made to date. That repo can be found at https://github.com/ESRM-Research/oqtane.docs. I can put the readable website version up on an Azure instance if that would help in reviewing them. As you can tell, there are still many unfinished pages, but the overall navigation is close to what I was planning for. Let me know if you would like to see any changes in the navigation or content. This is such a great project on many levels.

**Reply by iJungleboy** [reply]
@ESRM-Research Just fyi: I'm totally underwater till end of week, would love to review it after that.

**Reply by ESRM-Support** [reply]
@iJungleboy No problem. I still have quite a ways to go before I finish what I wanted to finish anyway. It has been a great way for me to learn the framework more fully. Hopefully by next week I can also post the results on an Azure website so it will be easier to review.

**Reply by ESRM-Support** [reply]
@iJungleboy I updated my GitHub repo with some further changes. Now at least all of the pages have some sort of explanation. I want to further refine many of the pages for consistency with the existing changes that I made and so forth. But right now I think that I should become more familiar with the Oqtane code and make sure that I am explaining things correctly. For example, the File Management isn't working as I would expect it to, so I am going to step through the code and see if I can understand what is going on. I would like to transition one of the major projects that I am working on to Oqtane by the end of the summer, so I am motivated to see what I need to do to be able to do that. 

I am not sure if you want to use what I have done to date or wait until later to look at it. It will probably be next week now before I can post what I have done so far on a website so it is easier to review. I will post the URL for that when that is ready. It will take me some additional weeks now before I get back to the documentation tasks since I need to get a deeper grasp of the framework. Hopefully that will be easier now that I understand at least some of the basics of Oqtane.

**Reply by MyLabBook** [reply]
@iJungleboy - I loaded my updated version of the Oqtane docs at the following URL. 

[http://oqtane.eastus.cloudapp.azure.com](http://oqtane.eastus.cloudapp.azure.com)

It is still very incomplete, but you can at least view some of the changes. I would be interested in what you think and what should be improved.

**Reply by MyLabBook** [reply]
@iJungleboy - Just as an update, I went ahead and got a domain so that I could make this site secure. You can now see this at https://oqdoc.com.

**Reply by sbwalker** [reply]
There is already a secure domain for docs located at https://docs.oqtane.org

**Reply by MyLabBook** [reply]
@sbwalker - Right, my thought was that was just a temporary place to make it easier for @iJungleboy to review my changes to see what he wanted to upload to https://docs.oqtane.org. I didn't know of another way to make it secure on Azure except to use a domain and I got a little paranoid about the site not being secure. But if you think it is too confusing, I can take it down.

**Reply by iJungleboy** [reply]
@ESRM-Research I would prefer it to be on the temporary domain, just to avoid any real public use. 

I see you've been changing a lot - structure and content. Are you just reorganizing, or also removing content? Because most of the content in Administration was contributed from a DNN book and we don't want to lose that. 

I would like to discourage the use of the colored title-images. The problem will be that if others contribute, they would start to create their own topic-header-images and this will very quickly look extremely inconsistent. So basically it causes a lot of overhead with little added value. 

![image](https://user-images.githubusercontent.com/5321186/173997238-7e467717-ccd0-4f3f-baa2-ed0db541f835.png)

Could you DM me on [info 2sxc.org](https://2sxc.org/en/contact) so we could organize a telco to chat through everything you are working on?

**Reply by MyLabBook** [reply]
@iJungleboy - Thanks for your feedback. I have taken down the https://oqdoc.com domain per your request. As mentioned, my preference is for SSL connections for security, so the other insecure temporary domain is no longer available. But it should be easy to recreate the docs locally from the forked repo given at the beginning if you think anything may be of use. 

As you mentioned, I reorganized a lot of the content, mostly just for me to get familiar with the framework and have a personal reference. I am really new to Oqtane, so the organization of the content is about all I can offer right now from a newbie standpoint. I did try to consolidate some of the Admin information to make it easier for me to navigate through, so some of that may be lost. So I am not sure how helpful those changes will be. It's probably going to take me a couple of months to get familiar with the code itself and offer any help in the more detailed documentation in the Admin area. 

You mentioned that some of the Admin material was based on a DNN book. If you could let me know what book is the best reference to use, then I can educate myself on that and perhaps provide some help down the road. 

I liked the use of color header images myself as a navigation tool, but I understand if you think that they are more trouble than they are worth. I sent you a message through your 2sxc.org website with my phone number if you want to talk more. But it may be that I need to become more educated about the project before I am able to help much. If you have any guidelines for what you are looking for in updating the documentation, though, that would be helpful. 

Thanks again for your feedback and time.

**Reply by iJungleboy** [reply]
@ESRM-Research In this case let's wait a bit before we make any changes. 

Because the docs should be maintained by people who have a broad understanding of what Oqtane is and does, otherwise changes would need so many reviews that it's too much work. 

I'll keep your phone number though and respond to the mail so you have my direct contact. 

In regards of the book, I believe we used https://www.amazon.com/Professional-DotNetNuke-Programming-Mitchel-Sellers/dp/0470171162

**Reply by MyLabBook** [reply]
@iJungleboy Yes, that makes sense. I ordered the book you referenced and plan to become much better informed about Oqtane over the coming months. Thanks for your time and feedback.

**Reply by sbwalker** [reply]
Some of the structure and content was based on this book: Professional DNN7: Open Source .NET CMS Platform https://a.co/d/8MSmCZ4 (I was the series editor for the DNN books from WROX Press and they transferred the copyright of this book to me). It's important to note that although there are some conceptual similarities between DNN and Oqtane there are a lot of differences... as Oqtane was built from the ground up as a new modular application framework based on modern technology and architecture. So it was really just some of the administrative chapters which were used in the documentation - the software development chapters would need to be completely rewritten for Oqtane.

**Reply by leigh-pointer** [reply]
> It's important to note that although there are some conceptual similarities between DNN and Oqtane there are a lot of differences

@sbwalker second that!

**Reply by MyLabBook** [reply]
@sbwalker Thanks for the updated book reference. I ordered that one as well. I will work on getting more informed on all the background. I recall looking at IBuySpy/DotNetNuke when it first came out and considered going down that road because the MS dev environment is so nice and the open source nature and relative ease of use of DotNetNuke . But I ended up going down the path of Drupal because it was fully open source. Now with .NET core being open source, it is time to return to having the best of it all - great dev environment, code, cloud, frameworks. So glad that you have committed to moving this work forward with a Blazor foundation.
